### ISSUE TITLE
聚类算法综述与信息检索顶会文章速读

### ISSUE THREAD

**Issue by wtysos11** [wtysos11/NoteBook#80]
信息检索大作业，打算作聚类算法的综述。大致的想法是介绍一下各个聚类算法及其代价，同时引用顶会上的文章。
IR的定会：SIGIR, KDD, WWW, NeurIPS, ICML, WSDM
PS:综合性：综述要纵横交错，既要以某一专题的发展为纵线，反映当前课题的进展；又要从本单位、省内、国内到国外，进行横的比较。只有如此，文章才会占有大量素材，经过综合分析、归纳整理、消化鉴别，使材料更精练、更明确、更有层次和更有逻辑，进而把握本专题发展规律和预测发展趋势。

资源：
* [剑桥大学IR课件-clustering](https://www.cl.cam.ac.uk/teaching/1314/InfoRtrv/lecture6.pdf)，里面主要介绍了hierarchical-clustering的主要算法HAC和non-hierarchical-clustering的主要算法K-means（此外还有k-median,k-center,correlation clustering等）
* [stanford-cluster in IR](https://nlp.stanford.edu/IR-book/html/htmledition/clustering-in-information-retrieval-1.html)
* [cluster领域的综述: Data clustering: 50 years beyond k-means](https://www.sciencedirect.com/science/article/pii/S0167865509002323)，我的总结在[这里](https://github.com/wtysos11/NoteBook/issues/81)
* [知乎-信息检索之聚类](https://zhuanlan.zhihu.com/p/32171153)，介绍了一些基础知识，相当于上面简介部分的中文版。

一些文章
* [Advanced Cluster Analysis](https://www.sciencedirect.com/science/article/pii/B9780123814791000113)

**Comment by wtysos11** [comment]
SIGIR 2019
# Sparse tensor co-clustering as a tool for document categorization
Abstract:
To deal with document clustering, we usually rely on document-term matrices. However, from additional available information like keywords, co-authors, citations we might rather exploit a reorganization of the data in the form of a tensor. In this paper, we extend the use of the Sparse Poisson Latent Block Model to deal with sparse tensor data using jointly all information arising from documents. The proposed model is parsimonious and tailored for this kind of data. To estimate the parameters, we derive a suitable tensor co-clustering algorithm. Empirical results on several real-world text datasets highlight the advantages of our proposal which improves the clustering results of documents.
内容：
在进行document clustering的时候一般会使用document-term matrices。问题：对于额外的一些信息，比如keywords, co-authors, citations我们可能会将数据重新组织为tensor，这个tensor是非常稀疏的。
使用Sparse Poisson Latent Block Model来处理这些稀疏的tensor data。Sparse Poisson Latent Block Model是基于Poisson distribution的document cluster方法。这是一个统计学的双聚类方法，消耗低，而且适用于稀疏矩阵。
为了处理数据，使用tensor co-clustering algorithm
实验结果由多个现实的文本数据集产生。
关键词：document clustering, co-clustering(双聚类/块聚类，同时对矩阵的行和列进行聚类的一种方法)
SIGIR 2017
# The Impact of Linkage Methods in Hierarchical Clustering for Active Learning to Rank

**Comment by wtysos11** [comment]
KDD 2019 
# Cluster-GCN: An Efficient Algorithm for Training Deep and Large Graph Convolutional Networks

**Comment by wtysos11** [comment]
KDD 2018
# Discovering Non-Redundant K-means Clusterings in Optimal Subspaces
A huge object collection in high-dimensional space can often be clustered in more than one way, for instance, objects could be clustered by their shape or alternatively by their color. Each grouping represents a different view of the data set. The new research field of non-redundant clustering addresses this class of problems. In this paper, we follow the approach that different, non-redundant k-means-like clusterings may exist in different, arbitrarily oriented subspaces of the high-dimensional space. We assume that these subspaces (and optionally a further noise space without any cluster structure) are orthogonal to each other. This assumption enables a particularly rigorous mathematical treatment of the non-redundant clustering problem and thus a particularly efficient algorithm, which we call Nr-Kmeans (for non-redundant k-means). The superiority of our algorithm is demonstrated both theoretically, as well as in extensive experiments.
A huge object collection in high-dimensional space can often be clustered in more than one way, for instance, objects could be clustered by their shape or alternatively by their color. Each grouping represents a different view of the data set. The new research field of non-redundant clustering addresses this class of problems. In this paper, we follow the approach that different, non-redundant k-means-like clusterings may exist in different, arbitrarily oriented subspaces of the high-dimensional space. We assume that these subspaces (and optionally a further noise space without any cluster structure) are orthogonal to each other. This assumption enables a particularly rigorous mathematical treatment of the non-redundant clustering problem and thus a particularly efficient algorithm, which we call Nr-Kmeans (for non-redundant k-means). The superiority of our algorithm is demonstrated both theoretically, as well as in extensive experiments.
高维空间上的大规模物体可以通过多于一种方式来聚类，举个例子，物体可以根据它们的颜色或形状来进行聚类。每一个类代表数据集合的一个不同的视角。一个新的研究领域是非重复聚类旨在解决这种问题。
在这篇文章中，我们使用的方法是不同的、无重复的、k-means-like的聚类算法，可能用于不同的、随机指向的高维空间中的子空间。我们假设这些子空间（可能存在没有任何聚类结构的噪音空间）互相正交，这个假设允许对非重复聚类问题比较严谨的数学计算，因此可以产生有效的算法，我们称其为Nr-Kmeans。
我们的算法不仅从理论上证明，而且有实际的实验
关键字：k-means，non-redundant

**Comment by wtysos11** [comment]
WWW 2019
# Learning Resolution Parameters for Graph Clustering
Finding clusters of well-connected nodes in a graph is an extensively studied problem in graph-based data analysis. Because of its many applications, a large number of distinct graph clustering objective functions and algorithms have already been proposed and analyzed. To aid practitioners in determining the best clustering approach to use in different applications, we present new techniques for automatically learning how to set clustering resolution parameters. These parameters control the size and structure of communities that are formed by optimizing a generalized objective function. We begin by formalizing the notion of a parameter fitness function, which measures how well a fixed input clustering approximately solves a generalized clustering objective for a specific resolution parameter value. Under reasonable assumptions, which suit two key graph clustering applications, such a parameter fitness function can be efficiently minimized using a bisection-like method, yielding a resolution parameter that fits well with the example clustering. We view our framework as a type of single-shot hyperparameter tuning, as we are able to learn a good resolution parameter with just a single example. Our general approach can be applied to learn resolution parameters for both local and global graph clustering objectives. We demonstrate its utility in several experiments on real-world data where it is helpful to learn resolution parameters from a given example clustering.

**Comment by wtysos11** [comment]
聚类算法的关键：
代表性点的选取
distance metrics 计算
object function
框架参考自http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.710.6923&rep=rep1&type=pdf
# cluster techniques for information retrieval

## 聚类算法的意义

[1]The definition of clustering is very different. Mainly it includes two part, placing similar objects in the same group and assign dissimilar objects to different groups(typically using a distance measure, such as Euclidean distance, etc.). This is a fundamental computational task, thereforre there is a large and still-growing suite of objective functions, algorithms ,and techniques for identifying good clusters.

Clustering is different from classification. Classification is supervised and requires a set of labelled training instances for each group, its input is a train set with label and a test set without label, and its output is a dataset with label.
Clustering is unsupervised and learns without a teacher to provide the labeling information of the training dataset.

## 按类别进行分类

There are two types of structures produced by clustering algorithms. Flat/ non-hierarchical clustering and hierarchical clustering
Flat clustering consists of a certain number of clusters and the relation between clusters is often undetermined. Its measurement is construction error minimization or probabilistic optimization.
Hierarchical clustering is a hierarchy with usual interpretation that each node stands for a sub-cluster of its mother's node. Its measurement is similarities of instances.

* flat clustering start out with a partition based on randomly selected seeds(one seed per cluster) and then refine the initial partition.
its algorithm: k-means and EM

### K-means
also caled Linde-Buzo-Gray in signal processing
It is hard clustering algorithm
Representation: Define clusters by the center of mass of their members
Objects should be represented in vector form.
其他视角：A kind of vector quantization, map from a continuous space to a discrete space 或者 color quantization, from 24 bits/pixel to 9 bits/pixel

距离判断标准为total reconstruction error 是 RSS: residual sum of squares，即每个点到各自聚类中心的距离之和

算法过程：略，自己看

复杂度：O(IKNM),I为迭代次数，K为聚类数量，N为对象数量，M为空间维度
K-means算法中初始化很重要。K-means不是最优化算法而是贪心算法，这意味着它并不一定取得最优解（虽然有证明，对于足够分离的聚类能够取得最优解）
Poor seeds will result in sub-optimal clustering

同类型：K-center, k-median等

### BFR算法
Bradley-Fayyad-Reina 对k-means的大数据特化算法

### CURE
Clustering Using REpresentatives

### EM算法
对EM算法有不错的讲解：https://www.whatmatrix.com/portal/clustering-algorithms-from-start-to-state-of-the-art-2/
A kind of model-based clustering
被看作是K-means的泛化
each cluster is a model for generating the data
* The centroid is a good representative for each model.
* Generate an object consisits of first picking a centroid at random and then adding some noise.

### Affinity Propagation

EM算法是Kmeans算法的soft version
It cluster as estimating a mixture of continuous probability distributions

Another important distinction between clustering algorithm is whether they perform soft or hard assignment.
Hard assignment means each object, or document in the context of IR, is assigned to one and only one cluster.
Soft assignment is probabilistic approach, it means each object may be assigned to multiple clusters.

Hierachical clustering usually adopts hard assignment, while in flat clustering, both types of assignments are common.

hierarchical clusteirng can be either bottom-up or top-down manners
* bottom-up(agglomerative)凝集的, start with individual objects and try to group the m,ost similar ones. The procedure terminates when one cluster containg all objects has been formed.
* top-dowm(divisive)分裂的,start with all objects in a group and divide them into groups so as to maximize within-group similarity

### HAC Hierarchical Agglomerative Clustering
A bottom-up approach
Assume a similarity measure for determining the similarity of two objects.
Start with all objects in a separate cluster and then repeatedly joins the two clusters that have the most similarity until there in only one cluster survived. The history of merging/clustering forms a binary tree or hierarchy.
Distance metrics and use euclidian distance, L1 Norm or Cosine Similarity

Measures of cluster similarity:
1. single-link clustering: The similarity between two clusters is the similarity of the two cloest objects in the cluster.(用相距最近的两个物体之间的距离作为聚类距离的替代) Search over all pairs of objects that are from the two different clusters and select the pair with the greatest similarity.
2. complete-link clustering: The similarity between two clusters is the similarity oof their two most dissimilar members(用最大距离代替) Its Preferable for most IR and NLP applications, but more sensitive to outliers.
3. group-average agglomerative clustering: a compromise between single-link and complete-link clustering. The similarity between two clusters is the average similarity between members.
4. centroid clustering. The similarity of two clusters is defined as the similarity of their centroids.

### Divisive clustering
A top-down approach
start with all objects in a single cluster
At each iteration, select the least coherent cluster and split it.
Continue the iteratoins until a predefined criterion is achieved. The history of clustering forms a binary tree or hierarchy.


## cluster质量

聚类内质量 internal criterion. The typical objective is to attain high intra-cluster similarity and low inter-cluster similarity. The measured quality depends on both the document representation and the similarity measure used.

external criterion evaluate how well the clustering matches the gold standard classes produced by human judges. That is, the quality is measured by the ability of the clustering algorithm to discover some or all of the hidden patterns or latent classes. 

Two common criteria: purity and rand index.

* purity: Each cluster is first assigned to class which is most frequent in the cluster. Then, the accuracy of the assignment is measured by counting the number of correctly assigned documents and dividing by the sample size. 每一个聚类中被正确分类的元素数量除以总数
* rand index measure the similarity between the clusters and the classes in ground truth. RI = (TP+TN)/total
* F-measure based on Rand index. P=TP/(TP+FP),R=TP/(TP+FN),Fb=(b^2+1)PR/(b^2P+R)
* NMI: Normalized Mutual information, has the same problem as the purity, does not penalize large cardinalities and thus does not formalize our bias.

## 按应用场景

result set clustering
scatter-gather
collection clustering
language modeling
cluster-based retrieval

## 最新的研究进展

[1]

其他关键词：

## co-clustering

Biclustering, block clustering, co-clustering or two-mode clustering is the same. It is a data mining technique which allows simultaneous clustering of the rows and colujmns of a matrix.(同时对矩阵的行和列进行聚类)

## subspace clustering
https://towardsdatascience.com/subspace-clustering-7b884e8fff73
面对高维数据
Subspace clustering is a technique which finds clusters within different subspace( a selection of one or more dimensions)
The underlying assumption is that we can find valid clusters which are defined by only a subset of dimensions.

The resulting clusters may be overlapping both in the space of features and observation.

## 多次聚类
Label aggregation for crowdsourcing with Bi-Layer clustering

## model-based clustering


## spectral clustering
新的聚类算法，从图论中演化出来的算法。
https://www.cnblogs.com/pinard/p/6221564.html
主成分分析
课件：https://people.csail.mit.edu/dsontag/courses/ml16/slides/lecture13.pdf

## graph clustering
问题的不同角度，以图的视角去看待聚类问题。
[arxiv - Document clustering using graph based document representation with constraints](https://arxiv.org/abs/1412.1888)
http://pages.di.unipi.it/marino/cluster18.pdf
Graph clustering refers to clustering of data in the form of graphs. 
Cluster graph, based on graph theory, is a graph formed from the disjoint union of complete graphs.

* Cut-based Measures
* Counting Measure


graph clustering 算法:http://web.iitd.ac.in/~bspanda/graphclustering.pdf
应用：https://arxiv.org/ftp/arxiv/papers/1412/1412.1888.pdf
### Minimum Spanning Tree based Clustering
k-spanning tree：对于最小生成树，移除k-1个权值最大的边

### Markov Clustering Algorithm and Random Walk
graph clustering 特有
simulate long random walk through graph
Random walks are calculated by Markov Chains

Venue category recommendation is an essential application for the tourism and advertisement industries, wherein it may suggest attractive localities within close proximity to users' current location. Considering that many adults use more than three social networks simultaneously, it is reasonable to leverage on this rapidly growing multi-source social media data to boost venue recommendation performance. Another approach to achieve higher recommendation results is to utilize group knowledge, which is able to diversify recommendation output. Taking into account these two aspects, we introduce a novel cross-network collaborative recommendation framework C3R, which utilizes both individual and group knowledge, while being trained on data from multiple social media sources. Group knowledge is derived based on new cross-source user community detection approach, which utilizes both inter-source relationship and the ability of sources to complement each other. To fully utilize multi-source multi-view data, we process user-generated content by employing state-of-the-art text, image, and location processing techniques. Our experimental results demonstrate the superiority of our multi-source framework over state-of-the-art baselines and different data source combinations. In addition, we suggest a new approach for automatic construction of inter-network relationship graph based on the data, which eliminates the necessity of having pre-defined domain knowledge.
界面推荐对于旅游业和广告业而言至关重要。考虑到许多人同时使用三个社交网络，考虑利用快速增长的多源多媒体数据来加速推荐算法是非常重要的。
